### PR TITLE
MODUSERBL-50 Implementing reset password link generation

### PR DESCRIPTION
**Pull request by dmytrotur**
MODUSERBL-50 Implementing reset password link generation
Link to JIRA issue: [MODUSERBL-50](https://issues.folio.org/browse/MODUSERBL-50)
Changes:

- new endpoint **POST /bl-users/password-reset/link** was implemented;
- updated interface version to v3.1 in module descriptior;
- updated module version to v4.1.0.

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+## 4.0.5
```
Comment (kostyantyn-kh) [inline NEWS.md]: Possibly new version could be 4.1.0 due to new endpoint implementation

Comment (dmytrotur) [reply NEWS.md]: Done

File: src/main/java/org/folio/service/PasswordRestLinkService.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package org.folio.service;
+
+import io.vertx.core.Future;
+import org.folio.rest.util.OkapiConnectionParams;
+
+/**
+ * Service for password reset link functionality
+ */
+public interface PasswordRestLinkService {
```
Comment (opochepam) [inline src/main/java/org/folio/service/PasswordRestLinkService.java]: seems like the name should be PasswordResetLinkService

Comment (dmytrotur) [reply src/main/java/org/folio/service/PasswordRestLinkService.java]: Done

**Pull request merged**